tests/phpunit/tests/html-api/wpHtmlTagProcessorModifiableText.php

```diff
@@ -0,0 +1,115 @@
+	 * @ticket {TICKET_NUMBER}
+	 *
+	 * @dataProvider data_tokens_not_supporting_modifiable_text_updates
+	 *
+	 * @param string $html             Contains HTML with a token not supporting modifiable text updates.
+	 * @param int    $advance_n_tokens Count of times to run `next_token()` before reaching target node.
+	 */
+	public function test_rejects_updates_on_unsupported_match_locations( string $html, int $advance_n_tokens ) {
+		$processor = new WP_HTML_Tag_Processor( $html );
+		while ( --$advance_n_tokens >= 0 ) {
+			$processor->next_token();
+		}
+
+		$this->assertFalse(
+			$processor->set_modifiable_text( 'Bazinga!' ),
+			'Should have prevented modifying the text at the target node.'
+		);
+
+		$this->assertSame(
+			$html,
+			$processor->get_updated_html(),
+			'Should not have modified the input document in any way.'
+		);
+	}
+
+	/**
+	 * Data provider.
+	 *
+	 * @return array[]
+	 */
+	private static function data_tokens_not_supporting_modifiable_text_updates() {
+		return array(
+			'Before parsing'               => array( 'nothing to see here', 0 ),
+			'After parsing'                => array( 'nothing here either', 2 ),
+			'Incomplete document'          => array( '<tag without="an end', 1 ),
+			'Presumptuous closer'          => array( 'before</>after', 2 ),
+			'Invalid (CDATA)'              => array( '<![CDATA[this is a comment]]>', 1 ),
+			'Invalid (shortest comment)'   => array( '<!-->', 1 ),
+			'Invalid (shorter comment)'    => array( '<!--->', 1 ),
+			'Invalid (markup declaration)' => array( '<!run>', 1 ),
+			'Invalid (PI-like node)'       => array( '<?xml is not html ?>', 1 ),
+		);
+	}
+
+	/**
+	 * Ensures that modifiable text updates are applied as expected to supported nodes.
+	 *
+	 * @ticket {TICKET_NUMBER}
+	 *
+	 * @dataProvider data_tokens_with_basic_modifiable_text_updates
+	 *
+	 * @param string $html             Contains HTML with a token supporting modifiable text updates.
+	 * @param int    $advance_n_tokens Count of times to run `next_token()` before reaching target node.
+	 * @param string $raw_replacement  This should be escaped properly when replaced as modifiable text.
+	 * @param string $transformed      Expected output after updating modifiable text.
+	 */
+	public function test_updates_basic_modifiable_text_on_supported_nodes( string $html, int $advance_n_tokens, string $raw_replacement, string $transformed ) {
+		$processor = new WP_HTML_Tag_Processor( $html );
+		while ( --$advance_n_tokens >= 0 ) {
+			$processor->next_token();
+		}
+
+		$this->assertTrue(
+			$processor->set_modifiable_text( $raw_replacement ),
+			'Should have modified the text at the target node.'
+		);
+
+		$this->assertSame(
+			$transformed,
+			$processor->get_updated_html(),
+			"Should have transformed the HTML as expected why modifying the target node's modifiable text."
+		);
+	}
+
+	/**
+	 * Data provider.
+	 *
+	 * @return array[]
+	 */
+	private static function data_tokens_with_basic_modifiable_text_updates() {
```

Comment:
There are some CI failures:

<img width="1424" alt="Screenshot 2024-07-12 at 09 36 20" src="https://github.com/user-attachments/assets/468fa4bb-1f8d-4bf3-bb06-799f7c84483b">

Does it have to be a public method maybe?

Reply:
yeah I've done this before on accident. made the data provider public in 0340ee3